File: 1497-design-a-stack-with-increment-operation/1497-design-a-stack-with-increment-operation.cpp
class CustomStack {
public:
    vector<int> stack;
    int top;
    int maxSize;  // Keeping track of the maximum size

    CustomStack(int maxSize) {
        stack.resize(maxSize); // Initialize the vector with maxSize capacity
        this->maxSize = maxSize;
        top = -1;             // Initialize top to -1 indicating an empty stack
    }

    void push(int x) {
        if (top + 1 < maxSize) { // Check if there's room to push
            top++;
            stack[top] = x;
        }
    }

    int pop() {
        if (top != -1) {
            return stack[top--];
        }
        return -1; // Return -1 if stack is empty
    }

    void increment(int k, int val) {
        for (int i = 0; i < min(k, top + 1); i++) {
            stack[i] += val;
        }
    }
};

File: 1497-design-a-stack-with-increment-operation/1497-design-a-stack-with-increment-operation.rs
struct CustomStack {
    stack: Vec<i32>,
    top: i32,
}

impl CustomStack {
    fn new(max_size: i32) -> Self {
        CustomStack {
            stack: vec![0; max_size as usize],
            top: -1,
        }
    }

    fn push(&mut self, x: i32) {
        if self.top < (self.stack.len() as i32 - 1) {
            self.top += 1;
            self.stack[self.top as usize] = x;
        }
    }

    fn pop(&mut self) -> i32 {
        if self.top != -1 {
            let value = self.stack[self.top as usize];
            self.top -= 1;
            return value;
        }
        -1
    }

    fn increment(&mut self, k: i32, val: i32) {
        for i in 0..std::cmp::min(k, self.top + 1) {
            self.stack[i as usize] += val;
        }
    }
}